modules/ROOT/pages/dataweave-language-guide.adoc

```diff
@@ -1,75 +1,212 @@
+
+For example, you can use the _single-value selector_ `payload.userName` to retrieve the single value `userName`, where `userName` represents an actual key in the input message.
+
+See xref:dataweave-selectors.adoc[Selectors] for a complete list of data selectors and their usage.
+
+=== Functions
+
+A function is a code block that contains a group of operations that are executed when the function is called in your application. Functions can accept different parameters as input that can then be used by the operations defined in the function. Additionally, every function returns a value after it finishes processing, and this value can then be used as input for a different function.
+
+Each DataWeave script that you write in any component (like a Transform component or Set Payload component) works as an independent program, meaning that all function definitions and calls are unique to the script defined in the component.
+
+See xref:dataweave-functions.adoc[Define DataWeave Functions] for additional function definition details.
+
+==== Function Definition vs. Function Call
+
+In DataWeave, you _define_ functions in the header of the script, creating a blueprint for the process that defines the operations that are executed upon the function call.
+
+When you _call_ a function from a DataWeave expression, you are instructing DataWeave to run all the operations defined in that function at that exact point of the script’s execution.
+
+For example, `upper` is a built-in function defined in the String module. You can call this function to convert a text value to uppercase. Also, you can use a selector to retrieve a value from the payload, and then execute the `upper()` function call with this value as the input parameter `upper(payload.someKey)`.
+
+DataWeave supports both _prefix_ notation (`function (param1, param2)`) and _infix_ notation (`param1 function param2`) for function calls. Note that you can use infix notation only with functions that accept only two parameters.
+
+==== Import Function Modules
+
+By default, DataWeave scripts import the `dw::Core` function module. To use functions from different modules, import the modules to your script by adding the `import` directive.
+
+See xref:dw-functions.adoc[DataWeave Reference] for instructions on using the `import` directive.
+
+=== Arrays
+
+An array is a data structure that can contain any number of elements of the supported DataWeave types. To declare an array, use brackets (`[` and `]`) to enclose a comma-separated list of elements, for example: `[1,2,3]`.
+
+Arrays also support the application of conditional logic to define conditional elements (`[(value1) if condition]`). For example, the expression `[(1) if true, (2) if false, (3) if payload != null]` defines the array `[1,3]` (assuming that `payload` is not `null`).
+
+=== Objects
+
+An object is a collection of zero or more `Key` and `value` pairs:
```

Comment:
We usually call these "key-value pairs" elsewhere, without the surrounding code format. Understood that the key belongs to a `Key` type. I'm not sure if we need the code formatting around the term "value".

Reply:
Actually, it's a bit inconsistent (`Key:value` or `key:value` pair when showing the format, "key-value pair" when talking about it generically): https://docs.mulesoft.com/mule-runtime/4.3/dataweave-types